src/cdk/v2/destinations/sprig/procWorkflow.yaml

```diff
@@ -0,0 +1,71 @@
+bindings:
+  - name: EventType
+    path: ../../../../constants
+  - path: ../../bindings/jsontemplate
+  - name: defaultRequestConfig
+    path: ../../../../v0/util
+  - name: removeUndefinedAndNullValues
+    path: ../../../../v0/util
+
+steps:
+  - name: messageType
+    template: |
+      .message.type.toLowerCase();
+
+  - name: validateInput
+    template: |
+      let messageType = $.outputs.messageType;
+      $.assert(messageType, "message Type is not present. Aborting");
+      $.assert(messageType in {{$.EventType.([.IDENTIFY, .TRACK])}}, "message type " + messageType + " is not supported");
+      $.assertConfig(.destination.Config.apiKey, "API Key is not present. Aborting");
+
+  - name: prepareIdentifyPayload
+    condition: $.outputs.messageType === {{$.EventType.IDENTIFY}}
+    template: |
+      $.context.payload = .message.({
+        userId: {{{{$.getGenericPaths("userIdOnly")}}}},
+        emailAddress: {{{{$.getGenericPaths("email")}}}},
+        attributes: .context.traits
+      });
+      $.context.payload = $.removeUndefinedAndNullValues($.context.payload);
+  
+  - name: validateIdentifyPayload
+    condition: $.outputs.messageType === {{$.EventType.IDENTIFY}}
+    template: |
+      $.assert($.context.payload.userId, "userId is required");
```

Comment:
This is redundant. Anyone of userId or anonymousId should be there otherwise the server will discard that event.

Reply:
yes, but we don't want to use anonymousId as anonymousId tracking is not supported here. so let's say if i started sending anonymousId as userId then in order to update it same anonymousId is needed else new user will get created, it will not get updated. only unique key here is userId.